libs/execution/src/lib/transforms/transform-executor.ts

```diff
@@ -63,37 +68,50 @@ export class TransformExecutor {
   }
 
   executeTransform(
-    column: TableColumn,
+    columns: Map<string, TableColumn>,
+    numberOfRows: number,
     context: ExecutionContext,
   ): {
     resultingColumn: TableColumn;
     rowsToDelete: number[];
   } {
     context.enterNode(this.transform);
 
-    const result = this.doExecuteTransform(column, context);
+    const result = this.doExecuteTransform(columns, numberOfRows, context);
     context.exitNode(this.transform);
 
     return result;
   }
 
   private doExecuteTransform(
-    column: TableColumn,
+    columns: Map<string, TableColumn>,
+    numberOfRows: number,
     context: ExecutionContext,
   ): {
     resultingColumn: TableColumn;
     rowsToDelete: number[];
   } {
-    const inputDetails = this.getInputDetails();
+    const inputDetailsList = this.getInputDetails();
     const outputDetails = this.getOutputDetails();
 
     const newColumn: Array<InternalValueRepresentation> = [];
     const rowsToDelete: number[] = [];
-    column.values.forEach((entry, rowIndex) => {
-      context.evaluationContext.setValueForReference(
-        inputDetails.port.name,
-        entry,
-      );
+
+    for (let rowIndex = 0; rowIndex < numberOfRows; ++rowIndex) {
+      // add variables to context
+      for (const inputDetails of inputDetailsList) {
+        const variableName = inputDetails.port.name;
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const column = columns.get(variableName)!;
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const variableValue = column.values[rowIndex]!;
```

Comment:
You could use `assert` statements instead of the `!` operator here to improve safety at runtime.